packages/manager/apps/web/client/app/hosting/cdn/shared/hosting-cdn-shared-settings.controller.js

```diff
@@ -0,0 +1,348 @@
+import set from 'lodash/set';
+import get from 'lodash/get';
+import find from 'lodash/find';
+import forEach from 'lodash/forEach';
+import assign from 'lodash/assign';
+import filter from 'lodash/filter';
+import maxBy from 'lodash/maxBy';
+import concat from 'lodash/concat';
+import map from 'lodash/map';
+import clone from 'lodash/clone';
+import includes from 'lodash/includes';
+
+import {
+  HOSTING_CDN_SHARED_SETTINGS_RULE_CACHE_RULE,
+  HOSTING_CDN_SHARED_SETTINGS_RULE_FACTOR_DAY,
+  HOSTING_CDN_SHARED_SETTINGS_RULE_FACTOR_HOUR,
+  HOSTING_CDN_SHARED_SETTINGS_RULE_FACTOR_MINUTE,
+  HOSTING_CDN_SHARED_SETTINGS_RULE_TYPE_BROTLI,
+  HOSTING_CDN_SHARED_SETTINGS_RULE_TYPE_DEVMODE,
+} from './hosting-cdn-shared-settings.constants';
+
+export default class CdnSharedSettingsController {
+  /* @ngInject */
+  constructor(
+    $q,
+    $scope,
+    $state,
+    $stateParams,
+    $translate,
+    $transitions,
+    $uibModal,
+    Alerter,
+    HostingCdnSharedService,
+  ) {
+    this.$q = $q;
+    this.$scope = $scope;
+    this.$state = $state;
+    this.$stateParams = $stateParams;
+    this.$translate = $translate;
+    this.$transitions = $transitions;
+    this.Alerter = Alerter;
+    this.HostingCdnSharedService = HostingCdnSharedService;
+
+    this.loading = { init: false };
+    this.params = {
+      domain: $stateParams.domain,
+      domainName: $stateParams.domainName,
+      serviceName: $stateParams.productId,
+    };
+    this.model = {
+      alwaysOnline: {
+        enabled: true,
+        changed: false,
+      },
+      http: {
+        enabled: true,
+        changed: false,
+      },
+      devmode: null,
+      brotli: null,
+      rules: [],
+    };
+    this.tasks = {
+      toAdd: [],
+      toUpdate: [],
+      toRemove: [],
+    };
+    this.HostingCdnSharedService.model = this.model;
+    this.HostingCdnSharedService.tasks = this.tasks;
+  }
+
+  $onInit() {
+    if (!this.params.domain) {
+      // this.goBack();
+      this.$state.go('app.hosting');
```

Comment:
Should be handled in state declaration, using `redirectTo` (see https://ui-router.github.io/ng1/docs/latest/interfaces/state.statedeclaration.html#redirectto)

Reply:
i fixed it.

Reply:
yes it great, but i cant because i am not able to inject a domain into it!

Reply:
fixed